pagespeed/envoy/envoy_cluster_manager.cc

```diff
@@ -107,13 +109,38 @@ void EnvoyClusterManager::initClusterManager() {
       dispatcher_->createDnsResolver({}), *ssl_context_manager_, *dispatcher_, *local_info_,
       secret_manager_, validation_context_, *api_, http_context_, *access_log_manager_,
       *singleton_manager_);
+}
 
-  Envoy::MessageUtil::loadFromFile(
-      "pagespeed/envoy/cluster.yaml",
-      bootstrap, Envoy::ProtobufMessage::getStrictValidationVisitor(), *api_);
-
-  cluster_manager_ = cluster_manager_factory_->clusterManagerFromProto(bootstrap);
+Envoy::Upstream::ClusterManager&
+EnvoyClusterManager::getClusterManager(const GoogleString str_url_) {
+  UriImpl uri(str_url_);
+  try {
+    uri.resolve(*dispatcher_, Envoy::Network::DnsLookupFamily::Auto);
+  } catch (UriException) {
+    // TODO : Error Handling.
+    std::cout << "UriException \n";
+    std::cout.flush();
+  }
+  cluster_manager_ =
+      cluster_manager_factory_->clusterManagerFromProto(createBootstrapConfiguration(uri));
   cluster_manager_->setInitializedCb([this]() -> void { init_manager_.initialize(init_watcher_); });
+  return *cluster_manager_;
+}
+
+const envoy::config::bootstrap::v2::Bootstrap
+EnvoyClusterManager::createBootstrapConfiguration(const Uri& uri) const {
```

Comment:
I guess the uri / utilities are all there just so you can pass Uri here right?
Let's just remove the argument, or pass in whatever we need for the relevant lines that need it:

```
  socket_address->set_address(uri.address()->ip()->addressAsString());
  socket_address->set_port_value(uri.port());
```

Probably, string address, uint32_t port or something like that suffices?
I think this is going to be temporary anyway, so we might as well just remove the arguments completely, and just hard code it for now until the forward proxy cluster type is introduced.

Reply:
used hardcoded argument i.e. Host, Port, and Scheme and pass it to createBootstrapConfiguration()
code pushed in https://github.com/apache/incubator-pagespeed-mod/pull/1969/commits/4ab87afdc150b403bcef176c8f4934e8f4ba18d1